### PR TITLE
Developer guide edits: rebase, cleanup, PRs, format

### PR DESCRIPTION
**Pull request by dagood**
Formatting:

* Title -> sentence case formatting from https://github.com/microsoft/go/pull/1417
* Add linebreaks for better reviewability and consistency.

A few changes:

* Mention `git go-patch rebase`.
* Be more specific about cleanup--under what circumstances there's a risk if you don't clean up.
* I don't think some of the text about patch files is necessary: patch files are explained early in the doc.
* Add more info about PRs.

### CODE REVIEW

File: eng/doc/DeveloperGuide.md [hunk 1]
```diff
@@ -165,19 +167,68 @@ Then, when you run:
 git go-patch extract
 ```
 
-The `go-patch` command will generate a patch file under the `go/patches` directory. The patch file will be prefixed with a serial number (one greater than the number of existing patch files), followed by a dash-separated commit message.
+The `go-patch` command will generate a patch file under the `go/patches` directory.
```
Comment (qmuntal) [inline eng/doc/DeveloperGuide.md]: For consistency:

```suggestion
The `go-patch extract` command will generate a patch file under the `go/patches` directory.
```

File: eng/doc/DeveloperGuide.md [hunk 2]
```diff
@@ -165,19 +167,68 @@ Then, when you run:
...
+The patch file will be prefixed with a serial number (one greater than the number of existing patch files), followed by a dash-separated commit message.
 
-### Squashing Changes to Existing Patch Files
+### Squashing changes to existing patch files
 
-Creating new patch files is not always necessary when there are existing patch files with similar purposes for the same files. In such cases, you can squash new commits on top of the existing ones to update their contents. The go-patch extract command will detect the differences in these commits and regenerate the patch files with the updated contents.
+Creating new patch files is not always necessary when there are existing patch files with similar purposes for the same files.
+In such cases, you can squash new commits on top of the existing ones to update their contents.
+The go-patch extract command will detect the differences in these commits and regenerate the patch files with the updated contents.
```
Comment (qmuntal) [inline eng/doc/DeveloperGuide.md]: For consistency:
```suggestion
The `go-patch extract` command will detect the differences in these commits and regenerate the patch files with the updated contents.
```

File: eng/doc/DeveloperGuide.md [hunk 3]
```diff
@@ -165,19 +167,68 @@ Then, when you run:
...
 
-Before starting work, please check the go/patches folder for any existing patch files related to the files you're working on. This helps maintain a clean repository by avoiding redundant patch files.
+Before starting work, please check the go/patches folder for any existing patch files related to the files you're working on.
```
Comment (qmuntal) [inline eng/doc/DeveloperGuide.md]: For consistency:
```suggestion
Before starting work, please check the `go/patches` directory for any existing patch files related to the files you're working on.
```

**Review by qmuntal — APPROVED** [review]
Approved with some minor comments.

File: eng/doc/DeveloperGuide.md (hunk 1 above)
Comment (dagood) [reply eng/doc/DeveloperGuide.md]: Hmm, didn't look at these, IMO they should all also include `git`.

**Review by qmuntal — APPROVED** [review]

**Review by mertakman — APPROVED** [review]

**Pull request merged**